### DISCUSSION TITLE
How do I use mobx-state-tree in my class components?

### DISCUSSION THREAD

**Discussion by tunm1228** [mobxjs/mobx-state-tree#2037] [Q&A] [answered]
**_Question_**

* [ ] I've checked documentation and searched for existing issues
* [ ] I tried the [spectrum channel](https://spectrum.chat/?t=dad48299-3dfc-4e10-b6da-9af1e39498a3)

**Comment by Ghirigoro** [comment]
What problem are you running into using MST with class components. Can you share an example of the issue you're running into. Also this might be a better fit for the Spectrum channel or stack overflow, since its not an issue with MST but a question as to its use.

**Comment by tunm1228** [comment]
import * as React from 'react';
import { View, ViewStyle, TextStyle, Image, TouchableOpacity } from 'react-native';
import { Screen, Text, Header } from "../../components"
import { color } from "../../theme"

export interface InforProps {
}

export interface InforState {
}

const ROOT: ViewStyle = {
    backgroundColor: '#fff',
    flex: 1
}

export default class InforComponent extends React.Component<InforProps, InforState> {
    constructor(props: InforProps) {
        super(props);
        this.state = {
        };
    }

    public render() {
        return (
            <Screen style={ROOT} preset="scroll" statusBar='dark-content'>
              
            </Screen>
        );
    }
}
yes this is my class created, i want to take out store from mobx but don't know how?

This is the package I use:
    "mobx-react": "^6.1.8",
    "mobx-react-lite": "^1.4.1",
    "mobx-state-tree": "^3.14.1",

**Comment by Ghirigoro** [comment, marked as answer]
I'm still not sure what your asking but the basic idea is that you would define your store model somewhere:

```
// store.js
import {types} from "mobx-state-tree";

const Store = types.model("Store", {
  time: types.string // or whatever it is you have in your store
}).actions(self=>({
  setTime(newTime) { self.time= Time; }
}));

export const store = Store.create();
```
And somewhere else you would have your React component class
```
// store-view.js
import * as React from 'react';
import { observer } from "mobx-react"
import { store } from "./store"

// You need observer so your component will react to changes in the store
@observer
export class StoreView extends React.Component {
  public render() {
    return (
    <div>
      <h1>LAST BUTTON PRESS TIME: { store.time }</h1>
      <button onClick={()=>store.setTime(date.now().toString())}>PRESS ME</button>
    </div>);
  }
}
```

**Comment by radiosilence** [comment]
@tunm1228 In order to avoid imports you should create a context provider with your store, then access the context in your components the normal way:

https://reactjs.org/docs/context.html

**Comment by coolsoftwaretyler** [comment]
Hey @tunm1228 - looks like it's been a while on this MST issue. Lots of good replies here. Did you find them helpful? If not, I'm happy to chat if this is still a relevant problem.

I'm going to convert this to a discussion, which will close the issue, but give you the option to mark helpful answers.

Thanks!

**Comment by coolsoftwaretyler** [comment]
Marking @Ghirigoro's answer as correct here since I think that's a pretty clear solution to the original question.
